api/src/passport.js

```diff
@@ -30,18 +32,39 @@ if (NODE_ENV === 'test') {
   }
 }
 else {
-  passport.serializeUser((user, done) => done(null, user))
-  passport.deserializeUser((id, done) => done(null, id))
+  passport.serializeUser((osmProfile, done) => {
+    done(null, osmProfile)
+  })
+
+  passport.deserializeUser((osmProfile, done) => {
+    done(null, osmProfile)
+  })
 
   passport.use(new OSMStrategy({
     consumerKey: OSM_CONSUMER_KEY,
     consumerSecret: OSM_CONSUMER_SECRET,
     callbackUrl: `${API_URL}/auth/openstreetmap/callback`
-  }, (token, tokenSecret, profile, done) => {
-    if (profile) {
-      return done(null, profile)
+  }, async (token, tokenSecret, profile, done) => {
```

Comment:
Should we be storing the token/tokenSecret? Would we be able to login using those so that we don't have to grant oauth access on every login? I'll look into the openstreetmap provider to try to better understand how that works.

Reply:
I think we can, so that we don't need to grant permissions to OSM every time. Right now if you logout and log back in, then you have to "grant permissions" again.